### PR TITLE
Addressing the issues identified in BZ 1341312

### PR DESCRIPTION
**Pull request by sferich888**
https://bugzilla.redhat.com/show_bug.cgi?id=1341312

### CODE REVIEW

File: images/router/haproxy/conf/haproxy-config.template [hunk 1]
```diff
@@ -230,7 +230,7 @@ backend be_edge_http_{{$cfgIdx}}
   {{ end }}
   http-request set-header Forwarded for=%[src];host=%[req.hdr(host)];proto=%[req.hdr(X-Forwarded-Proto)]
                 {{ range $idx, $endpoint := endpointsForAlias $cfg $serviceUnit }}
-  server {{$endpoint.IdHash}} {{$endpoint.IP}}:{{$endpoint.Port}} check inter 5000ms cookie {{$endpoint.IdHash}}
+  server {{$endpoint.IdHash}} {{$endpoint.IP}}:{{$endpoint.Port}} check inter {{env "ROUTER_BACKEN_CHECK_INTER" "5000ms"}} cookie {{$endpoint.IdHash}}
```
Comment (knobunc) [inline images/router/haproxy/conf/haproxy-config.template]: Whis is this BACKEN not BACKEND?

Comment (knobunc) [inline images/router/haproxy/conf/haproxy-config.template]: And should it be HAPROXY_ROUTER_BACKEND_CHECK_INTERVAL to be consistent with the SYSLOG variables?

**Comment by danmcp** [comment]
@ramr @knobunc Review please

**Comment by ramr** [comment]
@sferich888  could you just use a custom template for this, rather than do it in the default template.
You could still use the env variable in the custom template for the specific customer that needs this. 

The main reason am asking this is because this configuration knob for the check interval is a much better fit to use with route annotations. Since its more of a route specific configuration knob.
Basically add annotations to routes that need `"special handling"` rather than do it generally for all routes.

File: images/router/haproxy/conf/haproxy-config.template (hunk 1 above)
Comment (sferich888) [inline images/router/haproxy/conf/haproxy-config.template]: This should be fixed.

**Comment by sferich888** [comment]
@ramr what you say sounds like the correct approach, as I can see this setting needing to be different for different routes. 

My question are: 

1. What is the best way to go about reading / pulling the annotations
2. How do we account for annotations, that override defaults like: https://github.com/openshift/origin/pull/9518

I ask this because, we might want to make it a generic loop that adds configurations for routes, using annotations in a similar way to how https://github.com/docker/distribution/blob/master/docs/configuration.md#override-specific-configuration-options does this with env vars?

**Comment by ramr** [comment]
@sferich888  sorry for the delay replying - the best approach to pulling annotations is to
look at PR #9710 
As re: the second part, in general I would say env variables/config would be better suited for something that's global or we don't need on a per-route basis and annotations for something that we want to do on a per-route basis. We may have to revisit some of those env variables. And of course, there's always that fine line as to what knobs we want to provide and what can be done with a custom template.

**Comment by knobunc** [comment]
@sferich888 If you can rebase, I'll approve this.

**Comment by sferich888** [comment]
@knobunc should be re-based.

**Comment by JacobTanenbaum** [comment]
LGTM

**Comment by knobunc** [comment]
LGTM.  Thanks all.  [merge]
